### PR TITLE
Support yarn workspaces in build-tools

### PR DESCRIPTION
**Pull request by tylerbutler**
The build tools make assumptions that release groups (monorepos) are defined in lerna.json. As we consider moving to yarn, the build-tools need to load packages from different locations. This change adds logic into the common build-tools infrastructure to load packages from lerna.json if available, but to fall back to the `workspaces` field in package.json if lerna.json is missing.

The version for the release group will be loaded from lerna.json if present, otherwise the release group root package.json version will be used. This allows us to continue to version packages without using the root package.json to define it. That seems like it will be helpful, but I can't think of a specific use-case.

### CODE REVIEW

File: build-tools/packages/build-tools/src/common/monoRepo.ts [hunk 1]
```diff
@@ -33,26 +34,85 @@ function sentenceCase(str: string) {
...
+
+        if (pkgJson.workspaces !== undefined) {
+            console.log(`${kind}: Loading packages from ${packagePath}`);
+            for (const dir of pkgJson.workspaces as string[]) {
+                this.packages.push(...Packages.loadGlob(dir, kind, ignoredDirs, this));
+            }
+            return;
         }
-        this.version = lerna.version;
+        fatal(`Couldn't find lerna.json or package.json, or they were missing expected properties.`);
```
Comment (Josmithr) [inline build-tools/packages/build-tools/src/common/monoRepo.ts:115]: Nit: Might be nice to list the missing properties here to make the feedback more actionable if this is hit.

File: build-tools/packages/build-tools/src/common/npmPackage.ts [hunk 2]
```diff
@@ -277,6 +278,30 @@ export class Packages {
         return packages;
     }
 
+    public static loadGlob(globPath: string, group: MonoRepoKind, ignoredGlobs: string[] | undefined, monoRepo?: MonoRepo,): Package[] {
```
Comment (Josmithr) [inline build-tools/packages/build-tools/src/common/npmPackage.ts:281]: Docs? 😁

**Review by Josmithr — APPROVED** [review]

File: build-tools/packages/build-tools/src/common/monoRepo.ts (hunk 1 above)
Comment (tylerbutler) [reply build-tools/packages/build-tools/src/common/monoRepo.ts:115]: Yeah, it's more bookkeeping than I'd like to do at this point.

File: build-tools/packages/build-tools/src/common/npmPackage.ts (hunk 2 above)
Comment (tylerbutler) [reply build-tools/packages/build-tools/src/common/npmPackage.ts:290]: Added.

**Pull request merged**